### PR TITLE
Add Cart & Wishlist page with functionality

### PR DESCRIPTION
**Pull request by VLeads**
Added page cart & wishlist
Added functionality for cart & wishlist

Refractoring in auth part

(Edit: After login, please refresh once. There is some issue with the token, I will update code after submission as whole project need to be refractored)

Test Login: 
email: vishal@mail.com
password: Rapidstore1

### CODE REVIEW

File: src/components/card/cart-price-card/CartPriceCard.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { getNumbersInLocaleFormat } from "utils";
+
+export const CartPriceCard = ({ details }) => {
+  console.log("price-card", details);
```
Comment (BalajiHarinath) [inline src/components/card/cart-price-card/CartPriceCard.jsx]: Suggestion :- If not required remove the console log

File: src/components/card/cart-product-card/CartProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    deleteDataUsingApi: deleteWishlist,
+  } = useWishlist();
+
+  const {
+    data: wishlistData,
+    error: wishlistError,
+    isLoading: wishlistLoading,
+  } = wishlist;
+
+  console.log("cart", wishlistData, wishlistError, wishlistLoading);
```
Comment (BalajiHarinath) [inline src/components/card/cart-product-card/CartProductCard.jsx]: Remove console log

File: src/components/card/cart-product-card/cartProductCard.css [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+.cart-card-price-star{
+    display: flex;
+    justify-content: space-between;
+}
+
+.move-to-wishlist-btn{
+    margin-left: 1.6rem;
+    font-weight: inherit !important
```
Comment (BalajiHarinath) [inline src/components/card/cart-product-card/cartProductCard.css:42]: Suggestion : - Try if you can avoid !important

File: src/components/card/product-card/ProductCard.jsx [hunk 4]
```diff
@@ -1,54 +1,122 @@
...
+
+  const {
+    cart,
+    postDataUsingApi: postCartItem,
+    deleteDataUsingApi: deleteCartItem,
+  } = useCart();
+
+  const { data: cartData, error: cartError, isLoading: cartLoading } = cart;
+
+  console.log("cart", cartData);
```
Comment (BalajiHarinath) [inline src/components/card/product-card/ProductCard.jsx]: Remove console log

File: src/pages/cart/cart.css [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+.cart-container{
+    display: flex;
+    justify-content: space-evenly;
+}
+
+.cart-product-container{
+   
+  
+}
```
Comment (BalajiHarinath) [inline src/pages/cart/cart.css:11]: Suggestion:- You can remove this

File: src/pages/cart/Cart.jsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import { getPriceCardDetails } from "utils";
+
+export const Cart = () => {
+  const authToken = localStorage.getItem("token");
+
+  const { cart } = useCart();
+
+  const { data, isLoading, error } = cart;
+
+  console.log("cart", data);
```
Comment (BalajiHarinath) [inline src/pages/cart/Cart.jsx]: remove console log

File: src/utils/functions/getPriceCardDetails.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        acc.totalPrice +
+        getNumberFromFormattedPrice(curr.price.originalPrice) * curr.qty,
+      discountPrice:
+        acc.discountPrice +
+        curr.qty *
+          (getNumberFromFormattedPrice(curr.price.originalPrice) -
+            getNumberFromFormattedPrice(curr.price.ourPrice)),
+    }),
+    priceCardDetails
+  );
```
Comment (BalajiHarinath) [inline src/utils/functions/getPriceCardDetails.js:31]: Appreciation: Good that you have use a single reduce function to calculate both total price and discount price

**Review by Vishnu-Aithal — COMMENTED** [review]
Did not go through code yet , but while checking the deploy preview,
adding to cart and wishlist is failing with error code 500.

**Review by BalajiHarinath — APPROVED** [review]
Apart from the suggested changes everything looks good.

**Comment by VLeads** [comment]
> Did not go through code yet , but while checking the deploy preview, adding to cart and wishlist is failing with error code 500.

Actually this is working, there is a small bug, I'll figure it out later. You just have to reload page after login. And then it will work out. Please look at it.

**Comment by Vishnu-Aithal** [comment]
> > Did not go through code yet , but while checking the deploy preview, adding to cart and wishlist is failing with error code 500.
> 
> Actually this is working, there is a small bug, I'll figure it out later. You just have to reload page after login. And then it will work out. Please look at it.

Yes Its working after Reloading 👍

**Pull request merged**